File: backend/app/api/jobs.py
"""Job Criteria API routes"""
from fastapi import APIRouter, Depends, HTTPException, status
from sqlalchemy.orm import Session
from typing import List
from pydantic import BaseModel
from datetime import datetime

from app.core.dependencies import get_db
from app.models.models import JobCriteria

router = APIRouter(prefix="/jobs", tags=["jobs"])


class JobCriteriaCreate(BaseModel):
    title: str
    description: str
    recruiter_id: int = 1  # Default to admin for now


class JobCriteriaUpdate(BaseModel):
    title: str = None
    description: str = None


class JobCriteriaResponse(BaseModel):
    id: int
    recruiter_id: int
    title: str
    description: str
    created_at: datetime

    class Config:
        from_attributes = True


@router.get("/", response_model=List[JobCriteriaResponse])
def get_job_criteria(
    skip: int = 0,
    limit: int = 100,
    db: Session = Depends(get_db)
):
    """Get all job criteria"""
    criteria = db.query(JobCriteria).offset(skip).limit(limit).all()
    return criteria


@router.get("/{job_id}", response_model=JobCriteriaResponse)
def get_job_criteria_by_id(
    job_id: int,
    db: Session = Depends(get_db)
):
    """Get a specific job criteria by ID"""
    criteria = db.query(JobCriteria).filter(JobCriteria.id == job_id).first()
    if not criteria:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail="Job criteria not found"
        )
    return criteria


@router.post("/", response_model=JobCriteriaResponse)
def create_job_criteria(
    criteria: JobCriteriaCreate,
    db: Session = Depends(get_db)
):
    """Create a new job criteria"""
    db_criteria = JobCriteria(
        recruiter_id=criteria.recruiter_id,
        title=criteria.title,
        description=criteria.description
    )
    db.add(db_criteria)
    db.commit()
    db.refresh(db_criteria)
    return db_criteria


@router.put("/{job_id}", response_model=JobCriteriaResponse)
def update_job_criteria(
    job_id: int,
    criteria: JobCriteriaUpdate,
    db: Session = Depends(get_db)
):
    """Update a job criteria"""
    db_criteria = db.query(JobCriteria).filter(JobCriteria.id == job_id).first()
    if not db_criteria:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail="Job criteria not found"
        )
    
    if criteria.title:
        db_criteria.title = criteria.title
    if criteria.description:
        db_criteria.description = criteria.description
    
    db.commit()
    db.refresh(db_criteria)
    return db_criteria


@router.delete("/{job_id}", status_code=status.HTTP_204_NO_CONTENT)
def delete_job_criteria(
    job_id: int,
    db: Session = Depends(get_db)
):
    """Delete a job criteria"""
    criteria = db.query(JobCriteria).filter(JobCriteria.id == job_id).first()
    if not criteria:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail="Job criteria not found"
        )
    
    db.delete(criteria)
    db.commit()
    return None

File: frontend/src/services/jobs.ts
import apiClient from './api';

export interface JobCriteria {
  id: number;
  recruiter_id: number;
  title: string;
  description: string;
  created_at: string;
  criteria_skills?: Array<{ id: number; skill_id: number; weight: number }>;
}

export const jobsApi = {
  // Get all job criteria
  getJobs: (skip = 0, limit = 100) =>
    apiClient.get<JobCriteria[]>('/api/criteria/', { params: { skip, limit } }),

  // Get a specific job criteria
  getJob: (id: number) =>
    apiClient.get<JobCriteria>(`/api/criteria/${id}`),

  // Create a job criteria
  createJob: (data: { title: string; description: string; criteria_skills?: Array<{ skill_id: number; weight: number }> }) =>
    apiClient.post<JobCriteria>('/api/criteria/', data),

  // Update a job criteria
  updateJob: (id: number, data: Partial<JobCriteria>) =>
    apiClient.put<JobCriteria>(`/api/criteria/${id}`, data),

  // Delete a job criteria
  deleteJob: (id: number) =>
    apiClient.delete(`/api/criteria/${id}`),
};
